fastlane/lib/fastlane/environment_printer.rb

```diff
@@ -0,0 +1,207 @@
+module Fastlane
+  class EnvironmentPrinter
+    def self.get
+      env_output = ""
+      env_output << "### Fastlane Environment\n"
+      env_output << "\n"
+      env_output << print_system_environment
+      env_output << print_fastlane_files
+      env_output << print_loaded_fastlane_gems
+      env_output << print_loaded_plugins
+      env_output << print_loaded_gems
+
+      env_output
+    end
+
+    def self.print_loaded_plugins
+      ENV["FASTLANE_ENV_PRINTER"] = "enabled"
+      env_output =  "### Loaded Fastlane Plugins:\n"
+      env_output << "\n"
+      plugin_manager = Fastlane::PluginManager.new
+      plugin_manager.load_plugins
+      if plugin_manager.available_plugins.length <= 0
+        env_output << "**No Plugins Loaded***\n"
+      else
+        table = ""
+        table << "| Plugin | Version | Update-Status |\n"
+        table << "|--------|---------|\n"
+        plugin_manager.available_plugins.each do |plugin|
+          installed_version = Fastlane::ActionCollector.determine_version(plugin)
+          update_url = FastlaneCore::UpdateChecker.generate_fetch_url(plugin)
+          latest_version = FastlaneCore::UpdateChecker.fetch_latest(update_url)
+          if Gem::Version.new(installed_version) == Gem::Version.new(latest_version)
+            update_status = "😇 Up-To-Date"
+          else
+            update_status = "😱 Update availaible"
+          end
+          table << "| #{plugin} | #{installed_version} | #{update_status} |\n"
+        end
+
+        rendered_table = MarkdownTableFormatter.new table
+        env_output << rendered_table.to_md
+      end
+
+      env_output << "\n\n"
+      env_output
+    end
+
+    def self.print_loaded_fastlane_gems
+      # Fastlanes own gems
+      env_output = "### Fastlane Gems\n\n"
+      table = ""
+      table << "| Gem | Version | Update-Status |\n"
+      table << "|-----|---------|------------|\n"
+      Gem.loaded_specs.values.each do |x|
+        update_status = "N/A"
+
+        next unless Fastlane::Actions::UpdateFastlaneAction.all_tools.include?(x.name)
```

Comment:
That's not a good dependency to have. Instead you can just access `Fastlane::TOOLS`